Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -130,6 +131,7 @@ public function handle(CommandInterface $command): void
                     throw $concurrencyException;
                 }
             }
+            $this->commandHook->onAfterHandle($command);
```

Comment:
this line must be executed _after_ belows `if ($fullCatchUpResult->hadErrors()) {` as other-wise the cr is not up to date. Though im not sure if the hook should be run in the `finally` block if there was an exception like `$concurrencyException` ... probably not. So we need to change the `finally` block to be `catch (\Throwable)` and and run the `onAfterHandle` only if that was not the case.

Reply:
Yap ... I will take a deeper look. Just wanted to get it working first.

Reply:
Moved the hook to the end of the finally block. In case the `$fullCatchUpResult->hadErrors()` this is never reached because an error is thrown before. That way the behavior is equal regardless wether Events or a Generator are to be comitted.